packages/net/tcp_connection.pony

```diff
@@ -809,7 +809,20 @@ actor TCPConnection
     Resize the read buffer.
     """
     if _expect != 0 then
-      _read_buf.undefined(_expect.next_pow2().max(_next_size))
+      // We know how much data to expect. If _read_buf is too small,
+      // then resize it.
+      //
+      // _read_buf_offset may be very near the end of the array such
```

Comment:
this comment isn't quite correct for what will happen.

or i am misinterpreting it.

the realloc wont copy lots of what has already been consumed. the probably is the buffer is still huge with lots of available space BUT, we are reallocing anyway and quite possibly moving tons of data that we don't have to. if we want til the buffer is almost empty (or at least has less space then we need), then we will make fewer system calls.

Reply:
Agreed.  I have a change to the comments + change to `if (_read_buf.size() - _read_buf_offset) <= _expect then`, but I'll wait to force-push until the current crop of CI tests finish.

Reply:
bah, force push now.